proxy/contracts/schain/MessageProxyForSchain.sol

```diff
@@ -203,22 +216,35 @@ contract MessageProxyForSchain is MessageProxy, IMessageProxyForSchain {
         for (uint256 i = 0; i < messages.length; i++) {
             _callReceiverContract(fromChainHash, messages[i], startingCounter + 1);
         }
-        _topUpBalance();
+        _topUpSenderBalance();
     }
 
     /**
      * @dev Sets new version of contracts on schain
-     * 
+     *
      * Requirements:
-     * 
+     *
      * - `msg.sender` must be granted DEFAULT_ADMIN_ROLE.
      */
-    function setVersion(string calldata newVersion) external override {
-        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "DEFAULT_ADMIN_ROLE is required");
+    function setVersion(string calldata newVersion) external override onlyOwner {
         emit VersionUpdated(version, newVersion);
         version = newVersion;
     }
 
+    function setMinimumReceiverBalance(uint256 balance) external override onlyConstantSetter {
+        emit MinimumReceiverBalanceChanged(minimumReceiverBalance, balance);
+        minimumReceiverBalance = balance;
+    }
+
+    function topUpReceiverBalance(address payable receiver) external override {
+        require(isContractRegistered(bytes32(0), msg.sender), "Sender is not registered");
```

Comment:
Good!
I do not remember that we make a comment in the task, but did we decided that only "registeredForAll" contracts can top up or each "registered" contract

Reply:
It would be better to use this function `_authorizeOutgoingMessageSender`, to allow schain owner to add their custom contracts

Reply:
Discussed, will add this later